src/Symfony/Bundle/FrameworkBundle/Resources/config/message.xml

```diff
@@ -0,0 +1,70 @@
+<?xml version="1.0" ?>
+
+<container xmlns="http://symfony.com/schema/dic/services"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <services>
+        <defaults public="false" />
+
+        <!-- Bus -->
+        <service id="message_bus" class="Symfony\Component\Message\MessageBus" public="true">
+            <argument type="tagged" tag="message_middleware" /> <!-- Middlewares -->
+        </service>
+
+        <service id="Symfony\Component\Message\MessageBusInterface" alias="message_bus" />
+
+        <!-- Handlers -->
+        <service id="message.handler_resolver" class="Symfony\Component\Message\ContainerHandlerLocator">
+            <argument type="service" id="service_container"/>
+        </service>
+
+        <service id="message.middleware.call_message_handler" class="Symfony\Component\Message\Middleware\HandleMessageMiddleware">
+            <argument type="service" id="message.handler_resolver" />
+
+            <tag name="message_middleware" priority="-10" />
+        </service>
+
+        <!-- Asynchronous -->
+        <service id="message.asynchronous.routing.sender_locator" class="Symfony\Component\Message\Asynchronous\Routing\SenderLocator">
+            <argument type="service" id="message.sender_locator" />
+            <argument /> <!-- Message to sender ID mapping -->
+        </service>
+        <service id="message.asynchronous.middleware.send_message_to_producer" class="Symfony\Component\Message\Asynchronous\Middleware\SendMessageMiddleware">
+            <argument type="service" id="message.asynchronous.routing.sender_locator" />
+
+            <tag name="message_middleware" priority="-5" />
+        </service>
+
+        <!-- Message encoding/decoding -->
+        <service id="message.transport.serialize_message_with_type_in_headers" class="Symfony\Component\Message\Transport\Serialization\Serializer">
+            <argument type="service" id="serializer" />
+        </service>
+
+        <service id="message.transport.default_encoder" alias="message.transport.serialize_message_with_type_in_headers" public="true" />
+        <service id="message.transport.default_decoder" alias="message.transport.serialize_message_with_type_in_headers" public="true" />
+
+        <!-- Logging & Debug -->
+        <service id="message.middleware.debug.logging" class="Symfony\Component\Message\Debug\LoggingMiddleware">
+            <argument type="service" id="logger" />
```

Comment:
`logger` might not be available, the middleware definition needs to be removed in this case

Reply:
What if we do this? 
1. `LoggingMiddleware` supports `null` loggers
2. We remove this service if `!$container->getParameter('kernel.debug')`

Reply:
@chalasr what do you think?

Reply:
:+1: for removing it if not in debug mode. But if monolog is missing, given the whole class is about logging, I still think we should skip that middleware, and keep it depends on a logger

Reply:
Fixed in 3967edac2ff0a0a2e529b9ad005ee8bfc6340543.

Reply:
`logger` is always available since 3.4. If monolog isn't available, the minimal stderr logger is enabled.